test/test_sympy_utils.py

```diff
@@ -520,6 +523,74 @@ def test_z3_proof_floordiv_eq_simplify(self):
         r = solver.check()
         self.assertEqual(r, z3.unsat)
 
+class TestSingletonInt(TestCase):
+    def test_basic(self):
+        j1 = SingletonInt(1, 1)
+        j1_copy = SingletonInt(1, 1)
+        j2 = SingletonInt(2, 1)
+        j1x2 = SingletonInt(1, 2)
+
+        # eq
+        self.assertTrue(sympy.Eq(j1, j1))
+        self.assertTrue(sympy.Eq(j1, j1_copy))
+        self.assertFalse(sympy.Eq(j1, j2))
+        self.assertFalse(sympy.Eq(j1, j1x2))
+        self.assertFalse(sympy.Eq(j1, sympy.Integer(1)))
+        self.assertFalse(sympy.Eq(j1, sympy.Integer(3)))
+
+        # ne
+        self.assertFalse(sympy.Ne(j1, j1))
+        self.assertFalse(sympy.Ne(j1, j1_copy))
+        self.assertTrue(sympy.Ne(j1, j2))
+        self.assertTrue(sympy.Ne(j1, j1x2))
+        self.assertTrue(sympy.Ne(j1, sympy.Integer(1)))
+        self.assertTrue(sympy.Ne(j1, sympy.Integer(3)))
+
+        # ge
+        self.assertTrue(is_ge(j1, sympy.Integer(1)))
+        with self.assertRaisesRegex(ValueError, "indeterminate"):
+            is_ge(j1, sympy.Integer(3))
```

Comment:
Yeah this is a tricky one because sympy only allows overriding of Ge. The other relations (Lt, Gt, Le) are, by consequence, all derived from Ge e.g., Lt(a, b) := !Ge(a, b). This would mean that means that if we define the "indeterminate" j1 >= 3, the also "indeterminate" j0 < 3 will be evaluated to be True!

Reply:
yeah fine for now